mlflow/projects/__init__.py

```diff
@@ -534,8 +534,16 @@ def _get_docker_command(image, active_run, docker_args=None, volumes=None, user_
     cmd = [docker_path, "run", "--rm"]
 
     if docker_args:
-        for key, value in docker_args.items():
-            cmd += ['--' + key, value]
+        for name, value in docker_args.items():
+            # Passed just the name as boolean flag
+            if isinstance(value, bool) and value:
+                if len(name) == 1:
+                    cmd += ['-' + name]
```

Comment:
We should do this length check when `value` is not a bool as well. For example, docker's `-u/--user` argument requires a string value; the current implementation would produce `--u value` as opposed to `-u value`, causing an error.